File: Tawaf/app/src/main/java/me/mxcsyounes/tawaf/models/Point2D.java
package me.mxcsyounes.tawaf.models;

import android.graphics.PointF;

public class Point2D {

    private float x, y;

    public Point2D(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float getY() {
        return y;
    }

    public float getX() {
        return x;
    }

    public double distanceToLine(Line2D line) {
        double nominator = line.getA() * this.x + line.getB() * this.y + line.getC();
        double denominator = Math.sqrt(line.getA() * line.getA() + line.getB() * line.getB());
        return nominator / denominator;
    }

    public double distanceToPoint(Point2D point2D) {
        return Math.sqrt(this.square(this.getX() - point2D.getX()) +
                this.square(this.getY() - point2D.getY()));
    }

    private double square(double number) {
        return number * number;
    }
}

File: Tawaf/app/src/main/java/me/mxcsyounes/tawaf/models/Point2D.kt
package me.mxcsyounes.tawaf.models


data class Point2D(val x: Float, val y: Float) {

    // calculate distance to the point without abs value
    fun distanceToLine(line: Line2D): Double {
        val nominator = line.a * this.x + line.b * this.y + line.c
        val denominator = Math.sqrt((line.a * line.a + line.b * line.b).toDouble())
        return nominator / denominator
    }

    // distance from point to point
    fun distanceToPoint(point2D: Point2D): Double {
        return Math.sqrt(this.square((x - point2D.x).toDouble()) +
                this.square((y - point2D.y).toDouble()))
    }

    // simple square function
    private fun square(number: Double) = number * number

}